GW2EIEvtcParser/EIData/CombatReplay/Decorations/SkillModeDescriptor.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using GW2EIEvtcParser.ParsedData;
+using static GW2EIEvtcParser.ParserHelper;
+
+namespace GW2EIEvtcParser.EIData
+{
+    internal class SkillModeDescriptor 
+    {
+        [Flags]
+        public enum SkillModeCategory : uint
+        {
+            /// <summary>Unknown category</summary>
+            NotApplicable = 0,
+
+            /// <summary>Visible on player select</summary>
+            ShowOnSelect = 1 << 0,
+
+            /// <summary>Applies important buffs to allies</summary>
+            ImportantBuffs = 1 << 1,
+
+            /// <summary>Reflects or blocks projectiles</summary>
+            ProjectileManagement = 1 << 2,
+
+            /// <summary>Heals allies</summary>
+            Heal = 1 << 3,
+
+            /// <summary>Cleanses conditions from allies</summary>
+            Cleanse = 1 << 4,
+
+            /// <summary>Strips boons from enemies</summary>
+            Strip = 1 << 5,
+
+            /// <summary>Active portal</summary>
+            Portal = 1 << 6,
+        }
+
+        public AgentItem Owner;
+        public Spec Spec;
+        public long SkillID = 0;
+        public SkillModeCategory Category = SkillModeCategory.NotApplicable;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="owner">Owner of the skill, will use master if current is a minion</param>
+        /// <param name="spec">Spec of the skill, put Unknown for skills not specific to a certain spec</param>
+        /// <param name="skillID">ID of the skill</param>
+        /// <param name="category">Category of the skill</param>
+        /// <returns></returns>
+        public SkillModeDescriptor(AbstractSingleActor owner, Spec spec, long skillID = 0, SkillModeCategory category = SkillModeCategory.NotApplicable)
+        {
+            if (owner == null)
+            {
+                Owner = null;
```

Comment:
This could be a throw new InvalidOperation("SkillModeDescriptor must have an owner"), the old API had a null check as by putting owner null, you could later disable SkillMode on a decoration, with the description being on a different class, that is not necessary anymore. 
No reason to create such a class without an owner and if later in the code someone want to remove SkillMode from decoration they would simply do "decoration.UsingSkillMode(null)".